lib/route/link.c

```diff
@@ -1126,6 +1126,20 @@ static void link_keygen(struct nl_object *obj, uint32_t *hashkey,
 	return;
 }
 
+static int link_update(struct nl_object *old_obj, struct nl_object *new_obj)
+{
+	struct rtnl_link *src = (struct rtnl_link *) new_obj;
+	struct rtnl_link *dst = (struct rtnl_link *) old_obj;
+
+	if (!dst->l_af_ops)
+		return -NLE_OPNOTSUPP;
+
+	if (dst->l_af_ops->ao_update)
+		return dst->l_af_ops->ao_update(dst, src);
+
+	return -NLE_OPNOTSUPP;
```

Comment:
let's return early, so we handle failure (OPNOTSUPP) cases first. Also, add a code comment like:

```C
        struct rtnl_link *src = (struct rtnl_link *) new_obj;
        struct rtnl_link *dst = (struct rtnl_link *) old_obj;
        int r;

        if (!dst->l_af_ops)
		return -NLE_OPNOTSUPP;

	if (!dst->l_af_ops->ao_update)
		return -NLE_OPNOTSUPP;

        r = dst->l_af_ops->ao_update(dst, src);
        if (r < 0)
            return r;

        /* dst was already modified. We can no longer fail at this point. */

	return 0;
```